### PR TITLE
Adding slow op requests test

### PR DESCRIPTION
**Pull request by tintumathew10**
Signed-off-by: tintumathew10 <tmathew@redhat.com>

Adding the test for verifying slow op requests.

logs : http://magna002.ceph.redhat.com/ceph-qe-logs/tmathew/cephci-run-CR7SS7/

### CODE REVIEW

File: tests/rados/rados_test_util.py [hunk 1]
```diff
@@ -123,3 +123,23 @@ def get_device_path(host, osd_id):
     ][0]
     should_not_be_empty(dev_path, "Failed to get device path")
     return dev_path
+
+
+def get_slow_requests_log(installer, start, end):
+    """
+    Retrieve slow op requests from installer
```
Comment (pdhiran) [inline tests/rados/rados_test_util.py:130]: Can we add docstrings on the params sent, esp the formats in which `start and `end` need to be provided.

File: tests/rados/test_slow_op_requests.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        should_not_be_empty(acting_pg_set, "Failed to retrieve acting pg set")
+        osd_id = acting_pg_set[0]
+        host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        should_not_be_empty(host, "Failed to fetch host details")
+        dev_path = get_device_path(host, osd_id)
+        log.debug(
+            f"osd1 device path  : {dev_path}, osd_id : {osd_id}, host.hostname : {host.hostname}"
+        )
+
+        rados_obj.enable_file_logging()
```
Comment (pdhiran) [inline tests/rados/test_slow_op_requests.py:48]: I think since we are using `journalctl` to fetch logs, this would not be required.

**Review by pdhiran — APPROVED** [review]
Looks good . 


One observation : The entire test case being executed is very similar to other osd removal tests like ` tests/rados/test_osd_rebalance.py ` with the addition of checking logs via call : ` get_slow_requests_log(installer, start_time, end_time) `. In future, we can maybe try clubbing the execution of tests to save execution time.

File: tests/rados/test_slow_op_requests.py (hunk 2 above)
Comment (tintumathew10) [reply tests/rados/test_slow_op_requests.py:48]: When logging was not enabled, I was not able to fetch the slow op logs.

File: tests/rados/rados_test_util.py (hunk 1 above)
Comment (tintumathew10) [reply tests/rados/rados_test_util.py:130]: will be adding the doc string.

File: tests/rados/test_slow_op_requests.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        utils.set_osd_devices_unamanged(ceph_cluster, unmanaged=False)
+        rados_obj.change_recover_threads(config=pool, action="rm")
+        if config.get("delete_pools"):
+            for name in config["delete_pools"]:
+                method_should_succeed(rados_obj.detete_pool, name)
+            log.info("deleted all the given pools successfully")
+        return 0
+    except Exception as e:
+        log.info(e)
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply tests/rados/test_slow_op_requests.py:50]: done

File: tests/rados/test_slow_op_requests.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+from utility.utils import method_should_succeed, should_not_be_empty
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Perform OSD remove operation to verify slow op requests
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply tests/rados/test_slow_op_requests.py]: done.

File: tests/rados/rados_test_util.py [hunk 5]
```diff
@@ -123,3 +123,23 @@ def get_device_path(host, osd_id):
...
+    """
+    out = []
+    try:
+        d_out, d_err = installer.exec_command(
+            cmd="systemctl list-units --type=service | grep mon | grep ceph"
+        )
+        daemon = d_out.split(" ")[0]
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply tests/rados/rados_test_util.py]: done

File: suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml [hunk 6]
```diff
@@ -0,0 +1,118 @@
+# Suite contains tests related to osd re-balance upon OSD addition / removal
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml]: done

File: suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+      module: test_cephadm.py
+      polarion-id:
+      config:
+        verify_cluster_health: true
+        steps:
+          - config:
+              command: bootstrap
+              service: cephadm
+              base_cmd_args:
+                verbose: true
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml]: done

File: suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+      desc: Configures client admin node on cluster
+      module: test_client.py
+      polarion-id:
+      config:
+        command: add
+        id: client.1                      # client Id (<type>.<Id>)
+        node: node7                       # client node
+        install_packages:
+          - ceph-common
+          - ceph-base
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml]: done

File: tests/rados/rados_test_util.py [hunk 9]
```diff
@@ -123,3 +123,23 @@ def get_device_path(host, osd_id):
     ][0]
     should_not_be_empty(dev_path, "Failed to get device path")
     return dev_path
+
+
+def get_slow_requests_log(installer, start, end):
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply tests/rados/rados_test_util.py]: done

File: suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+      name: setup install pre-requisistes
+      desc: Setup phase to deploy the required pre-requisites for running the tests.
+      module: install_prereq.py
+      abort-on-fail: true
+
+  - test:
+      name: cluster deployment
+      desc: Execute the cluster deployment workflow.
+      module: test_cephadm.py
+      polarion-id:
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply suites/pacific/rados/tier-3_rados_test-slow-op-requests.yaml]: done

**Review by psathyan — COMMENTED** [review]
Looks good to me

**Review by SrinivasaBharath — APPROVED** [review]
LGTM

**Pull request merged**